src/pip/_internal/cache.py

```diff
@@ -245,13 +244,17 @@ def get(
                 # Built for a different python/arch/etc
                 continue
             candidates.append(
-                (wheel.support_index_min(supported_tags), wheel_name)
+                (
+                    wheel.support_index_min(supported_tags),
+                    wheel_name,
```

Comment:
nitpick: it's unlikely that the `min()` call would depend on the `wheel_name`.

Reply:
At least this is equivalent to the previous behavior. Alternatively, we may want to separately consider whether an sdist may produce a wheel with "build tag" that should be taken into account even in the cache.